### DISCUSSION TITLE
Couple of questions and thanks!

### DISCUSSION THREAD

**Discussion by boydkelly** [ohare93/brain-brew#28] [General]
Awesome plugin.   I'm thinking now that crowdanki and brainbrew should be integrated with Anki itself.   

I have had crowdanki installed for at least a year or longer, but never used as my need is not to collaborate using Anki, but to enable me to edit my cards in some other manner and sync back to Anki.   Editing in json is not really an option!!!   So brainbrew is really the answer to my requirements.   I will be able to migrate away from my tsv, awk scripts with Note-type aware exporter....   

Just wondering if there is a suggested directory structure for muliple decks.   I have created a brainbrew directory and exported via crowdanki to brainbrew/deck1 brainbrew/deck2 etc.   After running brainbrew init deck1/2, I get the build/deck1 and build/deck2 directories as will as recipes src etc.   But the recipes get overwritten as well as the src/data/*.csv files with any new deck.   Maybe I'm asking the question too soon without experimenting...   but do I *need* to create a separate directory for each deck with its own src, build, recipies etc?  Or is it possible or even preferable to keep that together by renaminng the recipies to be deck specific, and making some other config changes???

Also once I export the crowdanki file and do a brainbrew init, all the Anki sync will be happening with build/deck.json right?   So I should be able to nuke the initial crowdanki export file?   (I don't want to keep redundant data around)

Just an observation which is probably an issue with Anki itself.  When I did a test import from a build/deck json file (after runing the src to anki recipe, I got a couple of weird prompts in  (crowd)Anki, asking me to map note fields.  The weird part is it was referring to Note types that don't and haven't exisited in my Anki file for a couple of years!!   Something like Close*old that I may have renamed eons ago.   Where did that come from???   (retorical question)  Anyway my Anki file seems at this point to be ok.

Thanks again.   I'm sure this will become very useful for a lot of Anki users.

**Comment by jeprecated** [comment]
Hey mate, thank you for the kind words 👍 I am very happy to hear you are getting good use out of Brain Brew 👏 

> I'm thinking now that crowdanki and brainbrew should be integrated with Anki itself.

Good to hear 😊 (but will almost certainly never happen 😅) Though I would like to eventually incorporate Brain Brew as an Anki add-on.

> Just wondering if there is a suggested directory structure for muliple decks. I have created a brainbrew directory and exported via crowdanki to brainbrew/deck1 brainbrew/deck2 etc. After running brainbrew init deck1/2, I get the build/deck1 and build/deck2 directories as will as recipes src etc. But the recipes get overwritten as well as the src/data/*.csv files with any new deck. 

Sadly I am handicapped a lot by the structure of CrowdAnki at the moment (though I hope to change that in the future). The Brain Brew Init function is intended to get a user setup and running with one CrowdAnki export. It is important to mention **the init function does not do any work that a user cannot do themselves**. The structure of the deck, source files, and recipes are not rigid and can be changed 👍 You can change the name or location of any of the files, as long as you change the reference in the recipe / note model files. This includes joining two decks together by adding more csvs, note models, media folders, etc.

> Maybe I'm asking the question too soon without experimenting... but do I need to create a separate directory for each deck with its own src, build, recipies etc? Or is it possible or even preferable to keep that together by renaminng the recipies to be deck specific, and making some other config changes???

Currently Brain Brew only exports to one CrowdAnki deck, and I hope to add filters to allow cards of specific note type or with specific tags to go into specific decks, or vice versa that cards be taken from CrowdAnki export and put into specific csvs. What I do currently is sync an empty parent deck with child decks below it csvs using Brain Brew, then when I sync them back all the cards will be placed in the parent deck. However that can be mitigated by ticked "Do Not Move Existing Cards" when importing the deck back into Anki via CrowdAnki 👍 then all your new cards go into the parent folder (and can be sorted) and all the existing cards stay in their current location.

> Just an observation which is probably an issue with Anki itself. When I did a test import from a build/deck json file (after runing the src to anki recipe, I got a couple of weird prompts in (crowd)Anki, asking me to map note fields. The weird part is it was referring to Note types that don't and haven't exisited in my Anki file for a couple of years!! Something like Close*old that I may have renamed eons ago. Where did that come from??? (retorical question) Anyway my Anki file seems at this point to be ok.

Your CrowdAnki probably exported all note types you have, rather than just the ones used by the cards/deck you wanted exported. Check your Note Models in Anki and see if there are any that can be deleted 👍 

Thanks for the comments, mate.

**Comment by boydkelly** [comment]
You are welcome!   Actually what I would really like to do when brain-brew is ready is to sync my 2000 notes to individual markup or asciidoc files with a yaml header containing the note fields and data.   Then I could use one of the many notetaking systems from nvim, to obsidian or zettlr to add additional reasearch to the 'cards', and still sync the yaml header with anki.   I can even start to do this one-way right now with a little awk on the csv files >> notes-*.md.    I was checking out jq and yq json utilities, to operate from the deck.json file but I am not an expert with json and databases and haven't been able to get a 'filter" working to list the notes in yaml.   Please don't go out of your way... its off topic, but if you have any idea on what the filter is for jq < deck.json {filter}...  I would be interested.  Otherwise I can do what i need with awk and csv.   Thanks!

**Reply by boydkelly** [reply]
I did get something with jq / jq going.   It is very practical on the command line:
Enables you to pick and relabel whatever fields you need...

jq -r '[.notes[]|{guid:.guid,fr:.fields[0],dyu:.fields[3],tags:.tags}]' deck.json | yq -y > deck.yaml
